client/state/happychat/constants.js

```diff
@@ -1,2 +1,7 @@
+export const HAPPYCHAT_CONNECTION_ERROR_FORCED_CLOSE = 'forced close';
+export const HAPPYCHAT_CONNECTION_ERROR_PING_TIMEOUT = 'ping timeout';
+export const HAPPYCHAT_CONNECTION_ERROR_TRANSPORT_CLOSE = 'transport close';
+export const HAPPYCHAT_CONNECTION_ERROR_TRANSPORT_ERROR = 'transport error';
+
```

Comment:
These come from the [Socket.IO client library](https://github.com/socketio/socket.io-client/blob/master/dist/socket.io.js) — search for calls to `onClose(` and the passed-in values are the possible reasons for the connection error.

I'll add a comment here to document this better.